### PR TITLE
fix: modify signature of EmbargoApiClient.redirect_if_blocked

### PR DESCRIPTION
**Pull request by iloveagent57**
Make this signature match and use the same signature that
``openedx.core.djangoapps.embargo.redirect_if_blocked()`` now uses.

https://openedx.atlassian.net/browse/ENT-5777

A change was introduced in https://github.com/openedx/edx-platform/pull/30295 that modified the signature of `embargo.redirect_if_blocked()`, this change makes enterprise code respect that new signature.

**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `make static` has been run to update webpack bundling if any static content was updated
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [x] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

**Review by johnnagro — APPROVED** [review]
looks like an unused import can be cleaned up

File: CHANGELOG.rst [hunk 1]
```diff
@@ -17,6 +17,12 @@ Unreleased
 ----------
 None
 
+[3.46.0]
+--------
+fix: modify signature of EmbargoApiClient.redirect_if_blocked
+Make this signature match and use the same signature that
+``openedx.core.djangoapps.embargo.redirect_if_blocked()`` now uses.
```
Comment (adamstankiewicz) [inline CHANGELOG.rst:24]: nit: `openedx.core.djangoapps.embargo.api.redirect_if_blocked()`?

File: enterprise/views.py [hunk 2]
```diff
@@ -1786,7 +1785,10 @@ def get(self, request, enterprise_uuid, course_id):
         """
         # Check to see if access to the course run is restricted for this user.
         embargo_url = EmbargoApiClient.redirect_if_blocked(
-            [course_id], request.user, get_client_ip(request)[0], request.path)
+            request=request,
+            course_run_ids=[course_id],
```
Comment (adamstankiewicz) [inline enterprise/views.py:1789]: [clarification] Does `redirect_if_blocked` accept `course_run_ids`? It looks like it only accepts a single `course_key` kwarg?

```python
def redirect_if_blocked(
    request: Request,
    course_key: CourseKey,
    access_point: str = 'enrollment',
    user: Optional[types.User] = None,
) -> Optional[str]:
```

Comment (iloveagent57) [reply enterprise/views.py:1789]: phew, good catch.
I should try to manually test this.

Comment (iloveagent57) [reply enterprise/views.py:1789]: ah, wait, so - `EmbargoApiClient.redirect_if_blocked()` is really just a wrapper around the `embargo_api.redirect_if_blocked()` function.  If you look at the change in `lms.py`, we're using the `course_key` argument - our wrapper calls the real function in a loop, one iteration per course run id.

Comment (adamstankiewicz) [reply enterprise/views.py:1789]: Oh, yeah, my bad! The similar names of these methods got me conflated :)

**Review by adamstankiewicz — APPROVED** [review]

**Pull request merged**